### PR TITLE
	API endpoint to allow user to change password

### PR DESCRIPTION
**Pull request by DravitLochan**
Fixes issue #345 
Changes: added API end point for change password
test locally at http://127.0.0.1:4000/aaa/changepassword.json
send the following three parameters :
changepassword : contains email id of the user who's password has to be changed
password : current password
newpassword : new password

### CODE REVIEW

**Comment by DravitLochan** [comment]
@dynamitechetan @saurabhjn76 @madhavrathi @jyoth1raditya Please reivew!

**Comment by DravitLochan** [comment]
@dynamitechetan @saurabhjn76 @jyoth1raditya @Orbiter Please review!

File: src/ai/susi/server/api/aaa/PasswordChangeService.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    @Override
+    public ServiceResponse serviceImpl(Query post, HttpServletResponse response, Authorization rights, JsonObjectWithDefault permissions) throws APIException {
+        JSONObject result = new JSONObject();
+        result.put("accepted", false);
+
+        String useremail = post.get("changepassword", null);
+        String password = post.get("password", null);
+        String newpassword = post.get("newpassword",null);
```
Comment (sudheesh001) [inline src/ai/susi/server/api/aaa/PasswordChangeService.java]: Sending passwords to the server as plaintext doesn't seem to be the right thing to do. Can you send a hash and then decrypt the hash on the server if needed, ideally that shouldn't be done ever. Storing hashes and checking for hashes seems to be more promising.

But I think that is however a much larger pull request than the one being looked at here.

Comment (DravitLochan) [inline src/ai/susi/server/api/aaa/PasswordChangeService.java]: Yeah.! I too believe that sending password as plain text is a bit compromising with security of users, but same policy is followed in login, signup and reset password. So I believe if we have to change it, it needs to be done across all end points.

Comment (dynamitechetan) [inline src/ai/susi/server/api/aaa/PasswordChangeService.java]: Sending plain text passwords over https is quite secure, the traffic is encrypted. Of course, storing the passwords needs proper encryption,

**Pull request merged**